python_app/app.py

```diff
@@ -1,81 +1,82 @@
 import consul
-import requests
-from flask import Flask, jsonify, request
-from redisWrapper import RedisWrapper
-app = Flask(__name__)
 
 
-class ServiceDiscoverer():
-    def __init__(self, index=None, consistency='default'):
-        self.index = index
-        self.consistency = consistency
-        # TODO add ACL support
-        self.consul_conn = consul.Consul(host='consul_server', port='8500',
-                                         scheme='http')
+class ConsulConnection:
+    """
+    Generate connection to Consul
+    """
+    def __init__(self, host, port, scheme):
+        self.host = host
+        self.port = port
+        self.scheme = scheme
+
+    def connection(self):
+        return consul.Consul(host=self.host, port=self.port,
+                             scheme=self.scheme)
```

Comment:
```    def connection(self):
        return consul.Consul(host=self.host, 
                                            port=self.port,
                                            scheme=self.scheme)
```

Reply:
Yes I agree and that should be the return value of the init method, i suppose